lightning-rapid-gossip-sync/src/lib.rs

```diff
@@ -68,32 +74,55 @@ pub mod error;
 /// Core functionality of this crate
 pub mod processing;
 
-/// Sync gossip data from a file
-/// Returns the last sync timestamp to be used the next time rapid sync data is queried.
-///
-/// `network_graph`: The network graph to apply the updates to
-///
-/// `sync_path`: Path to the file where the gossip update data is located
-///
-pub fn sync_network_graph_with_file_path(
-	network_graph: &network_graph::NetworkGraph,
-	sync_path: &str,
-) -> Result<u32, GraphSyncError> {
-	let mut file = File::open(sync_path)?;
-	processing::update_network_graph_from_byte_stream(&network_graph, &mut file)
+/// Rapid Gossip Sync holder struct
+/// todo: ask reviewers for more interesting things to say here
+pub struct RapidGossipSync<NG: Deref<Target=NetworkGraph>> {
+	network_graph: NG,
+	is_initial_sync_complete: Arc<AtomicBool>
+}
+
+impl<NG: Deref<Target=NetworkGraph>> RapidGossipSync<NG> {
+	/// Instantiate a new RapidGossipSync holder
+	/// todo: same as above
```

Comment:
Wouldn't refer to this as a holder, FWIW.

Reply:
The code's outdated.